locator/gossiping_property_file_snitch.cc

```diff
@@ -12,33 +12,27 @@
 
 #include <seastar/core/file.hh>
 #include <seastar/core/seastar.hh>
+#include <seastar/coroutine/as_future.hh>
 #include "gms/versioned_value.hh"
 #include "gms/gossiper.hh"
 #include "utils/class_registrator.hh"
 
 namespace locator {
 future<bool> gossiping_property_file_snitch::property_file_was_modified() {
-    return open_file_dma(_prop_file_name, open_flags::ro)
-    .then([](file f) {
-        return do_with(std::move(f), [] (file& f) {
-            return f.stat();
-        });
-    }).then_wrapped([this] (auto&& f) {
-        try {
-            auto st = f.get();
-
+    auto f = co_await open_file_dma(_prop_file_name, open_flags::ro);
+    auto res = co_await coroutine::as_future(f.stat());
+    if (res.failed()) {
+            logger().error("Failed to open {} for read or to get stats", _prop_file_name);
+            co_await coroutine::return_exception_ptr(res.get_exception());
+    }
+            auto st = res.get();
```

Comment:
Is this transformation correct? To me it looks like the error handler branch will not react to exceptions thrown by `open_file_dma`.

I think it would be cleaner to use a `try..catch`, it would also result in a slightly simpler patch as you could reuse the existing `catch (...)` part. Not sure if it is a performance sensitive part of the code.

Reply:
It's not performance sensitive, but you're right, I lost the open_file_dma exceptions